source/agora/consensus/EnrollmentManager.d

```diff
@@ -146,10 +146,10 @@ public class EnrollmentManager
             /* preimages: */ PreImageCache(this.params.ValidatorCycle, 1)
         );
 
-        this.validator_set = new ValidatorSet(db_path, params);
-        this.enroll_pool = new EnrollmentPool(db_path);
-
         this.db = new ManagedDatabase(db_path);
+        this.validator_set = new ValidatorSet(this.db, params);
```

Comment:
Alternatively, maybe these should be separate files. But I'm not sure what's more ideal here.

Reply:
How about applying it in WAL(Write-Ahead Logging) option?
https://www.sqlite.org/wal.html

Reply:
It seems somewhat unrelated. It enables a reader and writer at the same time, but in our case we have multiple writers to the same file.

Anyway it's worth looking into WAL later, sure.